File: modules/network-image/android/src/main/java/expo/modules/networkimage/NetworkImageModule.kt
package expo.modules.networkimage

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NetworkImageModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("NetworkImage")

    View(NetworkImageView::class) {
      Prop("uri") { view: NetworkImageView, uri: String ->
        view.setUri(uri)
      }
      Prop("cornerRadius") { view: NetworkImageView, radius: Double ->
        view.setCornerRadius(radius.toFloat())
      }
      Prop("contentMode") { view: NetworkImageView, mode: String ->
        view.setContentModeString(mode)
      }
    }
  }
}

File: modules/network-image/ios/NetworkImageModule.swift
import ExpoModulesCore

public class NetworkImageModule: Module {
  public func definition() -> ModuleDefinition {
    Name("NetworkImage")

    View(NetworkImageView.self) {
      Prop("uri") { (view: NetworkImageView, uri: String) in
        view.setURI(uri)
      }

      Prop("cornerRadius") { (view: NetworkImageView, radius: Double) in
        view.setCornerRadius(CGFloat(radius))
      }

      Prop("contentMode") { (view: NetworkImageView, mode: String) in
        view.setContentModeString(mode)
      }
    }
  }
}
